settings/README.md

```diff
@@ -0,0 +1,238 @@
+the `package.json` file in the `jupyterlab` section (here `schema`):
+
+<!-- prettier-ignore-start -->
+```json5
+// package.json#L45-L48
+
+"jupyterlab": {
+  "extension": true,
+  "schemaDir": "schema"
+}
+```
+<!-- prettier-ignore-end -->
+
+And you should not forget to add it to the files of the package:
+
+```json5
+// package.json#L16-L20
+
+"files": [
+  "lib/**/*.{d.ts,eot,gif,html,jpg,js,js.map,json,png,svg,woff2,ttf}",
+  "schema/**/*.json",
+  "style/**/*.{css,eot,gif,html,jpg,json,png,svg,woff2,ttf}"
+],
+```
+
+Now that the settings are defined and included in the package, you can
+use it inside your extension. Let's look at this example:
+
+<!-- prettier-ignore-start -->
+```ts
+// src/index.ts#L17-L48
+
+activate: (app: JupyterFrontEnd, settings: ISettingRegistry) => {
+  let limit = 25;
+  let flag = false;
+
+  function loadSetting(setting: ISettingRegistry.ISettings) {
+    // Read the settings and convert to the correct type
+    limit = setting.get('limit').composite as number;
+    flag = setting.get('flag').composite as boolean;
+
+    console.log(`Limit is set to ${limit} and flag to ${flag}`);
+  }
+
+  // Wait for the application to be restored
+  app.restored
+    // Load the settings for this plugin
+    .then(() => settings.load(PLUGIN_ID))
+    .then(setting => {
+      // Read the settings
+      loadSetting(setting);
+
+      // Listen for settings changes using Signal
+      setting.changed.connect(loadSetting);
+
+      // Programmatically change a setting
+      return setting.set('limit', 20);
+    })
+    .catch(reason => {
+      console.error(
+        `Something went wrong when reading the settings.\n${reason}`
+      );
+    });
+}
+```
+<!-- prettier-ignore-end -->
+
+First, the extension waits for the application to be restored before
+loading the settings for your plugin:
+
+```ts
+// src/index.ts#L32-L32
+
+.then(() => settings.load(PLUGIN_ID))
+```
+
+Then once the settings are loaded, each setting can be read using
+the `get` method and the _setting id_ (the key defined in the settings
+JSON file). After getting the setting, you need to require the
+`composite` attribute to get its value and specify the type explicitly.
+
+```ts
+// src/index.ts#L21-L27
+
+function loadSetting(setting: ISettingRegistry.ISettings) {
+  // Read the settings and convert to the correct type
+  limit = setting.get('limit').composite as number;
+  flag = setting.get('flag').composite as boolean;
+
+  console.log(`Limit is set to ${limit} and flag to ${flag}`);
+}
+```
+
+To react at a setting change by the user, you should use the signal
+`changed`. In this case, when that signal is emitted the function
+`loadSetting` is called with the new settings.
+
+```ts
+// src/index.ts#L37-L38
+
+// Listen for settings changes using Signal
+setting.changed.connect(loadSetting);
+```
+
+Finally, in this example, a setting is changed programmatically. This
+use case can be interesting for example if some dialogs ask new values.
+
```

Comment:
Say something like `In this case, the initial value 25 for `limit` is set to `20` on the application restore (this includes the very first time the jupyterlab application is loaded by the user's browser).